packages/web3-eth/src/web3_eth.ts

```diff
@@ -1568,7 +1568,7 @@ export class Web3Eth extends Web3Context<Web3EthExecutionAPI, RegisteredSubscrip
 			subscription instanceof LogsSubscription &&
 			name === 'logs' &&
 			typeof args === 'object' &&
-			args.fromBlock &&
+			(args.fromBlock || args.fromBlock === 0) &&
```

Comment:
I think we should replace this with `!isNullish(args.fromBlock)`

Reply:
done